### PR TITLE
Add workspace in workspace test

### PR DESCRIPTION
**Pull request by LMG**

### CODE REVIEW

File: src/filter/mod.rs [hunk 1]
```diff
@@ -538,19 +538,56 @@ fn unapply2<'a>(
...
+                git2::Oid::zero(),
+                0o0100644,
+            )?;
+
+            // Insert a dummy file to prevent the directory from dissappearing through becoming
+            // empty.
+            let tree = tree::insert(
+                &transaction.repo(),
+                &tree,
+                &Path::new("DUMMY-df97a89d-b11f-4e1c-8400-345f895f0d40"),
```
Comment (LMG) [inline src/filter/mod.rs:556]: where does this number come from?

Comment (christian-schilling) [reply src/filter/mod.rs:556]: https://www.uuidgenerator.net/version4

Comment (christian-schilling) [reply src/filter/mod.rs:556]: It does not mean anything. just a filename that it unlikely to present in the tree by accident.

Comment (LMG) [reply src/filter/mod.rs:556]: ok

**Comment by LMG** [comment]
Look good to me, you can merge it

**Review by christian-schilling — APPROVED** [review]

**Pull request merged**